File: cpbook/2_data_structures/10487_closest_sums.py
# UVa 10487 - Closest Sums
# https://onlinejudge.org/external/104/10487.pdf

import sys

def test_case(t, n):
	a = [0] * n
	for i in range(n):
		a[i] = int(input())
	m = int(input())
	print("Case ", t, ":", sep="")
	for i in range(m):
		b = int(input())
		r = sys.maxsize
		for i in range(n):
			for j in range(i, n):
				if a[i] == a[j]:
					continue
				s = a[i] + a[j]
				if abs(s - b) < r:
					r = abs(s - b)
					t = s
		print("Closest sum to ", b, " is ", t, ".", sep="")


if __name__ == "__main__":
	t = 0
	while True:
		t += 1
		n = int(input())
		if n == 0:
			break
		test_case(t, n)

File: cpbook/2_data_structures/10487_closest_sums.cpp
// UVa 10487 - Closest Sums
// https://onlinejudge.org/external/104/10487.pdf

#include <iostream>
#include <climits>

using namespace std;

void test_case(int t, int n)
{
	int a[n];
	int i, j, m, b, r, s, d, e;

	for (i = 0; i < n; i++)	{
		cin >> a[i];
	}
	cin >> m;
	cout << "Case " << t << ":" << "\n";
	while (m--)	{
		cin >> b;
		r = INT_MAX;
		for (i = 0; i < n - 1; i++)	{
			for (j = i + 1; j < n; j++)	{
				s = a[i] + a[j];
				d = abs(s - b);
				if (d < r)	{
					r = d;
					e = s;
				}
			}
		}
		cout << "Closest sum to " << b << " is " << e << ".\n";
	}
}

int main()
{
	int t = 0, n;
	
	ios::sync_with_stdio(0);
	cin.tie(0);
	while(true)	{
		t++;
		cin >> n;
		if (n == 0)	{
			break;
		}
		test_case(t, n);
	}
	return 0;
}
		

